citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -379,4 +380,15 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
         }
         return result;
     }
-}
+
+    @Override
+    public ResponseEntity<String> deleteCitations(IUser user, String groupId, List<String> citationList) throws ZoteroConnectionException, ZoteroHttpStatusException {    
+        for(String element : citationList) {
+            Optional<ICitation> citation = citationStore.findById(element);
+            if(citation.isPresent()) {
+                citationStore.delete(citation.get());
+            }
+        }  
+        return zoteroManager.deleteMultipleItems(user, groupId, citationList, zoteroManager.getLatestGroupVersion(user, groupId));
```

Comment:
I would first delete the citations from Zotero and then locally. If something goes wrong with Zotero and items were not deleted the data is inconsistent otherwise.